iceoryx_posh/source/popo/ports/server_port_data.cpp

```diff
@@ -23,13 +23,17 @@ namespace popo
 {
 ServerPortData::ServerPortData(const capro::ServiceDescription& serviceDescription,
                                const RuntimeName_t& runtimeName,
-                               const NodeName_t& nodeName,
+                               const ServerOptions& serverOptions,
                                mepoo::MemoryManager* const memoryManager,
                                const mepoo::MemoryInfo& memoryInfo) noexcept
-    : BasePortData(serviceDescription, runtimeName, nodeName)
-    , m_chunkSenderData(memoryManager, SERVER_SUBSCRIBER_POLICY, 0, memoryInfo)
-    , m_chunkReceiverData(cxx::VariantQueueTypes::FiFo_MultiProducerSingleConsumer, SERVER_PUBLISHER_POLICY)
+    : BasePortData(serviceDescription, runtimeName, serverOptions.nodeName)
+    , m_chunkSenderData(
+          memoryManager, static_cast<SubscriberTooSlowPolicy>(serverOptions.clientTooSlowPolicy), 0, memoryInfo)
+    , m_chunkReceiverData(cxx::VariantQueueTypes::SoFi_MultiProducerSingleConsumer,
```

Comment:
It seems that `cxx::VariantQueueTypes::SoFi_MultiProducerSingleConsumer` and `QueueFullPolicy::DISCARD_OLDEST_DATA` are values which can maybe change in a safety critical environment to no discard data and a fifo. Would it therefore make sense to introduce some public constexpr member variables which state this somehow?

Reply:
Good question. We don't do this for pub-sub either. That does not mean we shouldn't do it but the question is if an additional constexpr will be of much use. I guess if we ever change this, it would be in the context of adding an additional field to `ServerOptions`.

Reply:
Could we add a type alias for the queue type and a constexpr for the behavior to `ServerOptions` for now and use them as default?

Reply:
I think I need to add this option to the server. Currently there is a `serverTooSlowPolicy` option for the client which does not make sense if there isn't a `requestQueueFullPolicy` on the server side. Will add this to the options but this can only be tested with integration tests and the discovery loop which connects the clients and servers.